lib/ex_doc/markdown/earmark.ex

```diff
@@ -24,22 +24,64 @@ defmodule ExDoc.Markdown.Earmark do
     * `:smartypants` - boolean. Turns on smartypants processing, so quotes become curly,
       two or three hyphens become en and em dashes, and so on. False by default
 
-    * `:highlight_tag` - the tag used for highlighting code, defaults to "span"
-
   """
-  def to_html(text, opts) do
+  @impl true
+  def to_ast(text, opts) do
     options =
       struct(Earmark.Options,
         gfm: Keyword.get(opts, :gfm, true),
         line: Keyword.get(opts, :line, 1),
-        file: Keyword.get(opts, :file),
+        file: Keyword.get(opts, :file, "nofile"),
         breaks: Keyword.get(opts, :breaks, false),
         smartypants: Keyword.get(opts, :smartypants, false),
         pure_links: true
       )
 
-    text
-    |> Earmark.as_html!(options)
-    |> ExDoc.Highlighter.highlight_code_blocks(opts)
+    case Earmark.as_ast(text, options) do
+      {:ok, ast, messages} ->
+        print_messages(messages, options)
+        fixup(ast)
+
+      {:error, ast, messages} ->
+        print_messages(messages, options)
+        ast
+    end
+  end
+
+  defp print_messages(messages, options) do
+    for {severity, line, message} <- messages do
+      file = options.file
+      IO.warn("#{inspect(__MODULE__)} (#{severity}) #{file}:#{line} #{message}", [])
+    end
+  end
+
+  defp fixup(list) when is_list(list), do: Enum.map(list, &fixup/1)
+  defp fixup(binary) when is_binary(binary), do: escape(binary)
+
+  @tags ~w(a br code h1 h2 h3 h4 h5 h6 ul ol li p pre img em strong table thead tbody tr th td blockquote)
+
+  defp fixup({tag, attrs, ast}) when tag in @tags do
+    attrs = Enum.map(attrs, &fixup_attr/1)
+    {String.to_atom(tag), attrs, fixup(ast)}
+  end
+
+  @attrs ~w(href class src alt style title)
+
+  defp fixup_attr({name, value}) when name in @attrs do
+    {String.to_atom(name), value}
+  end
+
+  # Credits: https://github.com/pragdave/earmark/blob/v1.4.2/lib/earmark/helpers.ex#L36:L54
+  # Should Earmark.as_ast/2 escape?
```

Comment:
Can you please open this up as an issue there?

Reply:
https://github.com/pragdave/earmark/issues/333